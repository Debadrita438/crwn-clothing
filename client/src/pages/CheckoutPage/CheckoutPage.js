import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { selectCartItems, selectCartTotal } from '../../redux/cart/cartSelector';

import CheckoutItem from '../../components/CheckoutItem/CheckoutItem';
import StripeButton from '../../components/StripeButton/StripeButton';
// import StripeCheckoutButton from '../../components/StripeButton/StripeCheckoutButton';

import { 
    CheckoutHeaderBlockContainer, 
    CheckoutHeaderContainer, 
    CheckoutPageContainer, 
    CheckoutTotal, 
    CheckoutWarning 
} from './CheckoutPage.styles';


const mapStateToProps = createStructuredSelector({
    cartItems: selectCartItems,
    total: selectCartTotal
});

const CheckoutPage = ({ cartItems, total }) => {
    return (
        <CheckoutPageContainer>
           <CheckoutHeaderContainer>
               <CheckoutHeaderBlockContainer>
                   <span>Product</span>
               </CheckoutHeaderBlockContainer>
               <CheckoutHeaderBlockContainer>
                   <span>Description</span>
               </CheckoutHeaderBlockContainer>
               <CheckoutHeaderBlockContainer>
                   <span>Quantity</span>
               </CheckoutHeaderBlockContainer>
               <CheckoutHeaderBlockContainer>
                   <span>Price</span>
               </CheckoutHeaderBlockContainer>
               <CheckoutHeaderBlockContainer>
                   <span>Remove</span>
               </CheckoutHeaderBlockContainer>
           </CheckoutHeaderContainer>
           {
               cartItems.map(cartItem => 
                    <CheckoutItem key={ cartItem.id } cartItem={ cartItem } />
                )
           }
           <CheckoutTotal>TOTAL: â‚¹{ total }</CheckoutTotal>
           <CheckoutWarning>
               *Please use the following test credit card for the payments*
               <br />
               Card No. - 4242 4242 4242 4242 (Visa Card)
               <br />
               Exp Date - 09/24 CVV - 123
           </CheckoutWarning>
           <StripeButton price={ total } />
           {/* <StripeCheckoutButton price={ total } /> */}
        </CheckoutPageContainer>
    );
}
 
export default connect(mapStateToProps)(CheckoutPage);