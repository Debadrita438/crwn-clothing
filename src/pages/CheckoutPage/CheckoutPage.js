import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { selectCartItems, selectCartTotal } from '../../redux/cart/cartSelector';

import CheckoutItem from '../../components/CheckoutItem/CheckoutItem';
import StripeButton from '../../components/StripeButton/StripeButton';
// import StripeCheckoutButton from '../../components/StripeButton/StripeCheckoutButton';

import './CheckoutPage.scss';


const mapStateToProps = createStructuredSelector({
    cartItems: selectCartItems,
    total: selectCartTotal
});

const CheckoutPage = ({ cartItems, total }) => {
    return (
        <div className='checkout-page'>
           <div className="checkout-header">
               <div className="header-block">
                   <span>Product</span>
               </div>
               <div className="header-block">
                   <span>Description</span>
               </div>
               <div className="header-block">
                   <span>Quantity</span>
               </div>
               <div className="header-block">
                   <span>Price</span>
               </div>
               <div className="header-block">
                   <span>Remove</span>
               </div>
           </div>
           {
               cartItems.map(cartItem => 
                    <CheckoutItem key={ cartItem.id } cartItem={ cartItem } />
                )
           }
           <div className="total">TOTAL: â‚¹{ total }</div>
           <div className='test-warning'>
               *Please use the following test credit card for the payments*
               <br />
               Card No. - 4242 4242 4242 4242 (Visa Card)
               <br />
               Exp Date - 09/24 CVV - 123
           </div>
           <StripeButton price={ total } />
           {/* <StripeCheckoutButton price={ total } /> */}
        </div>
    );
}
 
export default connect(mapStateToProps)(CheckoutPage);