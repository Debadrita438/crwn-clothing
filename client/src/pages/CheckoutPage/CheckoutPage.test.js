import { shallow } from 'enzyme';

import { CheckoutPage } from './CheckoutPage';

describe('CheckoutPage', () => {
    let wrapper;

    beforeEach(() => {
        const mockProps = {
            cartItems: [],
            total: 0
        };

        wrapper = shallow(<CheckoutPage { ...mockProps } />);
    });

    it('should render CheckoutPage', () => {
        expect(wrapper).toMatchSnapshot();
    });
});