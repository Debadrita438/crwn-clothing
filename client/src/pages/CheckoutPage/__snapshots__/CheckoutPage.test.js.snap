// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutPage should render CheckoutPage 1`] = `ShallowWrapper {}`;
